In [2]:
from scipy.stats import chisquare

observed = [21307, 21306, 21387]
expected = [64000 / 3] * 3

result = chisquare(f_obs=observed, f_exp=expected)

print(f"Chi-square statistic: {result.statistic:.6f}")
print(f"P-value: {result.pvalue:.6f}")

Chi-square statistic: 0.202531
P-value: 0.903693


### SRM Test Results

- Chi-square statistic: 0.202531
- Degrees of freedom: 2
- P-value: 0.903693
- Significance level: 0.05

**Interpretation:** We fail to reject the null hypothesis.
There is no statistically significant evidence of a sample
ratio mismatch under the assumed 1:1:1 allocation.


### Covariate Balance Assessment

Pre-treatment customer characteristics were compared across
the three randomized groups using standardized mean
differences (SMDs).

| Covariate | Men's Email vs Control | Women's Email vs Control |
|---|---:|---:|
| History | 0.0076 | 0.0065 |
| Recency | 0.0068 | 0.0052 |
| Newbie | -0.0009 | 0.0026 |
| Men's merchandise history | -0.0046 | -0.0086 |
| Women's merchandise history | 0.0076 | 0.0049 |

**Interpretation:** All evaluated covariates have absolute
SMDs below 0.01, indicating negligible observed
standardized differences between treatment and control groups.

Together with the SRM test, these findings provide no
obvious evidence of allocation imbalance in the evaluated
variables. They do not establish that randomization was
perfect or that all potential imbalances have been ruled out.

**Limitation:** Balance checks for categorical covariates
such as channel and zip_code remain to be completed.
